### ISSUE TITLE
AST differs for combination of Haddocks and data decls

### ISSUE THREAD

**Issue by amesgen** [tweag/ormolu#827]
**Describe the bug**
The snippet
```haskell
data Foo
  -- | A
  = A -- ^ a
  -- | B
  | B
```
is formatted into
```haskell
data Foo
  = -- | A
    A -- a
    -- | B
  | B
```
which has differing AST (in the input, the parser considers `-- | B` to be a continuation of the invalid Haddock comment `-- ^ a`, but in the output, `-- | B` is a valid Haddock comment). 

**Environment**
0.3.1.0 and e003e11a569f9801b2a3f0cc01a66c883d597a3d

**Additional context**
Found via #821 (in `compiler/GHC/Driver/Types.hs`).

**Comment by mrkkrp** [comment]
This doesn't seem to happen anymore. Currently, the snippet is transformed into:

```haskell
data Foo
  = -- | A
    A -- \^ a
    -- \| B
  | B
```

@amesgen should we close this issue?

**Comment by amesgen** [comment]
Yes, it was fixed as a side effect of #837, thanks for noticing!
